components/ai_chat/common/mojom/ai_chat.mojom

```diff
@@ -55,11 +61,13 @@ interface PageHandler {
   OpenBraveLeoWiki();
   DisconnectPageContents();
   ClearConversationHistory();
+  RetryAPIRequest();
 };
 
 interface ChatUIPage {
   OnConversationHistoryUpdate();
   OnAPIRequestInProgress(bool in_progress);
+  OnAPIResponseError(APIError error);
```

Comment:
Only having an event but not providing a state property means that switching away from the Conversation and then switching back again won't show the error state, right? I think that's a problem if so since the user might not see the error.

Reply:
makes sense. I have added a getter and initiating it like others on initialiseTab